packet_packer.go

```diff
@@ -331,9 +331,14 @@ func (p *packetPacker) PackCoalescedPacket(onlyAck bool, maxPacketSize protocol.
 		initialHdr, initialPayload = p.maybeGetCryptoPacket(maxPacketSize-protocol.ByteCount(initialSealer.Overhead()), protocol.EncryptionInitial, onlyAck, true, v)
 		if initialPayload.length > 0 {
 			size += p.longHeaderPacketLength(initialHdr, initialPayload, v) + protocol.ByteCount(initialSealer.Overhead())
+
+			if size >= maxPacketSize-protocol.MinCoalescedPacketSize {
```

Comment:
This if doesn’t seem correct:
1. You’re not allowed to add a PING if `ackOnly` is set.
2. You should only add a PING if there’s no other frames in the packet.

Reply:
> You should only add a PING if there’s no other frames in the packet.

is it a bug in our code ? because I confirmed this from [rfc9000#section-17.2.2-8](https://datatracker.ietf.org/doc/html/rfc9000#section-17.2.2-8) which says 

> The payload of an Initial packet includes a CRYPTO frame (or frames) containing a cryptographic handshake message, ACK frames, or both. PING, PADDING, and CONNECTION_CLOSE frames of type 0x1c are also permitted.

Reply:
Why do you think it’s a bug? All these frames are permitted, but we’re only concerned with the situation where we only happen to have an ACK frame to send.

Reply:
so it means we only need to check these conditions `!onlyAck && len(initialPayload.frames) == 0` 
because `initialPayload.ack` can be there or cannot be there.

Reply:
@marten-seemann you got the chance to review this PR again ?

Reply:
@marten-seemann is this PR good to merge ?